src/gtc/passes/oir_optimizations/horizontal_execution_merging.py

```diff
@@ -363,22 +363,17 @@ def visit_VerticalLoop(self, node: oir.VerticalLoop, **kwargs: Any) -> oir.Verti
         )
 
     def visit_Stencil(self, node: oir.Stencil, **kwargs: Any) -> oir.Stencil:
-        vertical_loops: List[oir.VerticalLoop] = []
         protected_fields = set(n.name for n in node.params)
-        all_names = collect_symbol_names(node)
-        vertical_loops = [
-            *reversed(
-                [
-                    self.visit(
-                        vl,
-                        new_symbol_name=symbol_name_creator(all_names),
-                        protected_fields=protected_fields,
-                        **kwargs,
-                    )
-                    for vl in reversed(node.vertical_loops)
-                ]
+        new_symbol_name = symbol_name_creator(collect_symbol_names(node))
+        vertical_loops = []
```

Comment:
I'm surprised that mypy doesn't enforce typing the `vertical_loops` variable. If it works, great!